src/bv2int.cpp

```diff
@@ -312,7 +304,16 @@ Term BV2Int::handle_bw_op(const Term & t,
     y = tmp;
   }
 
-  Term res = get_explicit_bw(op, bv_width, TermVec({x, y}));
+  Term res;
+  if (lazy_bw_) {
+    // in lazy mode, don't want to add anything to extra_assertions_
+    // pass a throwaway vector instead
+    TermVec dummy_vec;
+    res = utils_.gen_bw(op, bv_width, granularity_, x, y, dummy_vec);
```

Comment:
Sorry for being a nag about this, but for large bitwidths and large terms this could become a bottleneck I think. Better not to bother with the extra assertions at all, that is, doing this check in gen_bw, or maybe even better, having a Boolean flag as an extra argument to `gen_bw`, as u c fit.

Reply:
No that's a good point! I added a helper function for getting the UF and in lazy mode I only call that. What do you think of that? Done in 5303d59d5156ffedb9e15170d751fcfde3609d1d